TidepoolKit/Model/Datum/TCGMSettingsDatum.swift

```diff
@@ -0,0 +1,240 @@
+        try container.encodeIfPresent(outOfRangeAlertsDEPRECATED, forKey: .outOfRangeAlertsDEPRECATED)
+        try container.encodeIfPresent(rateOfChangeAlertsDEPRECATED, forKey: .rateOfChangeAlertsDEPRECATED)
+        try super.encode(to: encoder)
+    }
+
+    public struct Alerts: Codable, Equatable {
+        public var enabled: Bool?
+        public var urgentLow: LevelAlert?
+        public var urgentLowPredicted: LevelAlert?
+        public var low: LevelAlert?
+        public var lowPredicted: LevelAlert?
+        public var high: LevelAlert?
+        public var highPredicted: LevelAlert?
+        public var fall: RateAlert?
+        public var rise: RateAlert?
+        public var noData: DurationAlert?
+        public var outOfRange: DurationAlert?
+
+        public init(enabled: Bool, urgentLow: LevelAlert? = nil, urgentLowPredicted: LevelAlert? = nil, low: LevelAlert? = nil, lowPredicted: LevelAlert? = nil, high: LevelAlert? = nil, highPredicted: LevelAlert? = nil, fall: RateAlert? = nil, rise: RateAlert? = nil, noData: DurationAlert? = nil, outOfRange: DurationAlert? = nil) {
+            self.enabled = enabled
+            self.urgentLow = urgentLow
+            self.urgentLowPredicted = urgentLowPredicted
+            self.low = low
+            self.lowPredicted = lowPredicted
+            self.high = high
+            self.highPredicted = highPredicted
+            self.fall = fall
+            self.rise = rise
+            self.noData = noData
+            self.outOfRange = outOfRange
+        }
+
+        public struct DurationAlert: Codable, Equatable {
+            public enum Units: String, Codable {
+                case hours
+                case minutes
+                case seconds
+            }
+
+            public var enabled: Bool?
+            public var duration: Double?
+            public var units: Units?
+            public var snooze: Snooze?
+
+            public init(enabled: Bool, duration: Double, units: Units, snooze: Snooze? = nil) {
+                self.enabled = enabled
+                self.duration = duration
+                self.units = units
+                self.snooze = snooze
+            }
+        }
+
+        public struct LevelAlert: Codable, Equatable {
+            public typealias Units = TBloodGlucose.Units
+
+            public var enabled: Bool?
+            public var level: Double?
+            public var units: Units?
+            public var snooze: Snooze?
+
+            public init(enabled: Bool, level: Double, units: Units, snooze: Snooze? = nil) {
+                self.enabled = enabled
+                self.level = level
+                self.units = units
+                self.snooze = snooze
+            }
+        }
+
+        public struct RateAlert: Codable, Equatable {
+            public enum Units: String, Codable {
+                case milligramsPerDeciliterPerMinute = "mg/dL/minute"
+                case millimolesPerLiterPerMinute = "mmol/L/minute"
```

Comment:
Could this not just use the units of the CGM settings and add `/minute`?

Reply:
In theory, yes, but that would require a complete `Codable` implementation for `RateAlert` to support that here. Not sure that is worth the effort.